src/backend/distributed/commands/function.c

```diff
@@ -157,3 +161,95 @@ EnsureSequentialModeForFunctionDDL(void)
 						 "use only one connection for all future commands")));
 	SetLocalMultiShardModifyModeToSequential();
 }
+
+
+/*
+ * ShouldPropagateAlterFunction returns, based on the address of a function, if alter
+ * statements targeting the function should be propagated.
+ */
+static bool
+ShouldPropagateAlterFunction(const ObjectAddress *address)
```

Comment:
I think such a check should come in the hook after we have checked we should propagate.
We should throw an error telling the `SET` command is unsupported for citus. This might need to be replicated to the distribution of the function as well, otherwise one could set the unsupported values before distributing and get it set that way.

For now I would not care and apply them all without any checks.